include/dlaf/eigensolver/tridiag_solver.h

```diff
@@ -55,7 +56,11 @@ void tridiagSolver(Matrix<BaseType<T>, device>& tridiag, Matrix<BaseType<T>, dev
   DLAF_ASSERT(tridiag.distribution().size().rows() == evals.distribution().size().rows(),
               tridiag.distribution().size().rows(), evals.distribution().size().rows());
 
+#if defined(DLAF_WITH_HIP)
+  DLAF_UNIMPLEMENTED("Tridiagonal solver is not yet implemented for HIP");
```

Comment:
This was required because the linker checks for symbols when linking shared libraries. The full eigensolver is instantiated which instantiates a call to the tridiagonal solver and the tridiagonal solver is not instantiated for HIP. This is a temporary solution like https://github.com/eth-cscs/DLA-Future/commit/95bedc17d2d50a605bf50e104534f1e6ce2e001a.